hvac/api/auth_methods/kerberos.py

```diff
@@ -0,0 +1,485 @@
+#!/usr/bin/env python
+"""Kerberos API methods module."""
+import fnmatch
+import logging
+import os
+import string
+from os.path import isfile
+from random import random
+from urllib.parse import urlparse
+
+import gssapi
+import gssapi.raw as gssapi_raw
+from gssapi.raw.misc import GSSError
+from requests_gssapi import HTTPSPNEGOAuth
+from requests_gssapi.exceptions import SPNEGOExchangeError
+
+from hvac import utils
+from hvac.api.auth_methods.ldap import Ldap as LdapAuth
+from hvac.exceptions import Unauthorized
+
+DEFAULT_MOUNT_POINT = "kerberos"
+HOST_TEMPLATE = '_HOST'
+ENV_KRB5_CONF = 'KRB5_CONFIG' # (optional) environment variable tells us the path to your krb5.conf file.
+ENV_KRB5_CACHE = 'KRB5CCNAME'  # (optional) environment variable tells us the directory of your kerberos credential cache
+SPNEGO_OID = '1.3.6.1.5.5.2'
+
+
+class KerberosError(Exception):
+    """
+    Please keep this exception so users can differentiate between kerberos issues and general api errors.
+    """
+    pass
+
+
+class Kerberos(LdapAuth):  # TODO for discussion do you want ldap subclassed or to be verbose and a bit redundant ?
```

Comment:
> `# TODO for discussion do you want ldap subclassed or to be verbose and a bit redundant ?`

I think we want to be verbose/redundant.

We don't know if/how the LDAP API will change, and would like these to be independent.

If we were going to look to DRY up things like this, I think we'd make some sort of abstract class one level higher, and have it be the parent for both of these (and anything else it applies to) but for right now I think we ought to duplicate.